src/pcre2_jit_compile.c

```diff
@@ -1271,6 +1270,9 @@ while (cc < ccend)
     cc++;
     break;
 
```

Comment:
I think this also needs on top (for safety):

```c
#if !defined(SUPPORT_UNICODE) || PCRE2_CODE_UNIT_WIDTH == 8
case OP_XCLASS:
#endif
```

Reply:
I removed the ifdefs for cases where they were used to remove code that wasn't needed.

@PhilipHazel would you prefer:

```c
  case OP_ECLASS:
  case OP_XCLASS:
  return cc + GET(cc, 1);
```

or:

```c
  case OP_ECLASS:
#ifdef SUPPORT_WIDE_CHARACTERS
  case OP_XCLASS:
#endif
  return cc + GET(cc, 1);
```

It seems a bit excessive use ifdefs just to remove the "case", when the code underneath is required regardless.

Reply:
the point I was making is that if the JIT compiler ever gets an OP_XCLASS in the 8 bit library without utf (which before couldn't happen because it was not even defined) it might break, so it is safer to detect that case here as unsupported IMHO.

Reply:
Oh I see. That's fair; I'll add the check you describe to check_opcode_types().